org.eclipse.winery.frontends/app/tosca-management/src/app/instance/sharedComponents/properties/properties.component.html

```diff
@@ -11,33 +11,105 @@
   ~
   ~ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~-->
-<div *ngIf="!loading; else loadingIcon">
-    <div class="right">
-        <button [disabled]="!sharedData?.currentVersion?.editable || properties === null"
-                class="btn btn-primary"
-                (click)="save()">
-            Save
-        </button>
-    </div>
-    <div *ngIf="propertyKeys?.length > 0; else elseBlock">
-        <table>
-            <tr *ngFor="let prop of propertyKeys">
-                <td>
-                    <label class="propertyLabel" for="{{ prop }}">{{ prop }}</label>
-                </td>
-                <td>
-                    <input class="propertyInput" id="{{ prop }}" [(ngModel)]="properties[prop]">
-                </td>
-            </tr>
-        </table>
-    </div>
-    <ng-template #elseBlock>
-        <div *ngIf="properties === null">
-            The type does not have a “Properties Definition”.
-        </div>
-        <winery-editor #propertiesEditor *ngIf="isXMLData" [ngModel]="properties"></winery-editor>
-    </ng-template>
+<div *ngIf="!isLoading(); else loadingIcon">
+    
+    <ng-container *ngIf="!definitions.length">
+        <alert type="info">The type does not have any "Properties Definitions".</alert>
+    </ng-container>
+
+    <ng-container *ngIf="definitions.length">
+
+        <ng-container *ngIf="isXML">
+
+            <div class="right">
+                <button [disabled]="!instanceService?.currentVersion?.editable || properties === null"
+                        class="btn btn-primary"
+                        (click)="save()">
+                    Save
+                </button>
+            </div>
+
+            <winery-editor #propertiesEditor [ngModel]="properties"></winery-editor>
+        </ng-container>
+
+        <ng-container *ngIf="!isXML">
+            <form #form="ngForm" (ngSubmit)="save()" novalidate>
+
+                <div class="right">
+                    <button type="submit" class="btn btn-primary" [disabled]="!form.valid">Save</button>
+                </div>
+
+                <br/>
+                <br/>
+
+                <div class="form-group">
+
+                    <ng-container *ngFor="let definition of definitions; index as index">
+
+
```

Comment:
There are a lot of empty lines, I suggest to remove them, it makes it not more readable - on the contrary...